lib/textrazor/request.rb

```diff
@@ -21,18 +16,59 @@ class Request
       classifiers: 'classifiers'
     }
 
-    def self.post(text, options)
-      ::RestClient.post url, build_query(text, options), accept_encoding: 'gzip'
+    def self.post(text, api_key:, **options)
+      ApiClient.post(
+        '',
+        build_query(text, options),
+        'x-textrazor-key': api_key, accept_encoding: 'gzip'
+      )
+    end
+
+    def self.create_dictionary(dictionary, api_key:, **options)
+      ApiClient.put(
+        "entities/#{dictionary.id}",
+        dictionary.to_h.to_json,
+        'x-textrazor-key': api_key, accept_encoding: 'gzip'
+      )
+      dictionary
+    end
+
+    def self.get_dictionary_entries(dictionary_id, limit:, offset:, api_key:)
+      ApiClient.get(
+        "entities/#{dictionary_id}/_all?limit=#{limit}&offset=#{offset}",
+        'x-textrazor-key': api_key, accept_encoding: 'gzip'
+      )
+    end
+
+    def self.delete_dictionary(dictionary_id, api_key:)
+      ApiClient.delete(
+        "entities/#{dictionary_id}",
+        'x-textrazor-key': api_key, accept_encoding: 'gzip'
+      )
+      true
+    end
+
+    def self.create_dictionary_entries(dictionary_id, dictionary_entries, api_key:)
+      ApiClient.post(
+        "entities/#{dictionary_id}/",
+        dictionary_entries.map(&:to_h).to_json,
+        'x-textrazor-key': api_key, accept_encoding: 'gzip'
+      )
+      dictionary_entries
     end
 
-    def self.url
-      TextRazor.configuration.secure ? HTTPS_URL : HTTP_URL
+    def self.delete_dictionary_entry(dictionary_id, dictionary_entry_id, api_key:)
+      ApiClient.delete(
+        "entities/#{dictionary_id}/#{dictionary_entry_id}",
+        'x-textrazor-key': api_key, accept_encoding: 'gzip'
+      )
+      true
     end
 
     private
 
     def self.build_query(text, options)
-      query = {"text" => text, "apiKey" => options.delete(:api_key)}
+      query = {"text" => text}
```

Comment:
Removing it from here means this is not duplicated across 4 or 5 different methods. Any particular reason to do this?

Reply:
The API key must be passed as a `x-textrazor-key` header.

As of today, this is not an issue because the only endpoint supported by the gem is a `POST` request, but adding support for dictionaries means supporting `GET`, `PUT` and `DELETE` requests.